CMakeLists.txt

```diff
@@ -0,0 +1,35 @@
+cmake_minimum_required(VERSION 3.27)
+
+project(PythonQt LANGUAGES CXX VERSION 3.5.6)
+
+set(CMAKE_CXX_STANDARD 11)
+set(CMAKE_CXX_STANDARD_REQUIRED ON)
+
+find_package(QT NAMES Qt6 Qt5 REQUIRED COMPONENTS Core)
+find_package(Python3 COMPONENTS Development)
+
+set(PYTHONQT_SUFFIX Qt${QT_VERSION_MAJOR}.${QT_VERSION_MINOR}-Python${Python3_VERSION_MAJOR}.${Python3_VERSION_MINOR})
+
+
+if(${QT_VERSION_MAJOR} VERSION_EQUAL 5)
+    if(${QT_VERSION_MINOR} VERSION_LESS 3)
+        set(PYTHONQT_GENERATED_PATH ${CMAKE_CURRENT_LIST_DIR}/generated_cpp_50)
+    elseif(${QT_VERSION_MINOR} VERSION_LESS 6)
+        set(PYTHONQT_GENERATED_PATH ${CMAKE_CURRENT_LIST_DIR}/generated_cpp_54)
+    elseif(${QT_VERSION_MINOR} VERSION_LESS 11)
+        set(PYTHONQT_GENERATED_PATH ${CMAKE_CURRENT_LIST_DIR}/generated_cpp_56)
+    elseif(${QT_VERSION_MINOR} VERSION_LESS 15)
+        set(PYTHONQT_GENERATED_PATH ${CMAKE_CURRENT_LIST_DIR}/generated_cpp_511)
+    else()
+        set(PYTHONQT_GENERATED_PATH ${CMAKE_CURRENT_LIST_DIR}/generated_cpp_515)
+    endif()
+# elseif(${QT_VERSION_MAJOR} VERSION_EQUAL 6)
+else()
+    message(FATAL "No generated sources exist for Qt${QT_VERSION}")
```

Comment:
This does not handle the (actually preferred) case of self-generated wrappers in `generated_cpp`, as done in the qmake file. Especially this will not work for Qt6, unless I'm missing something.

Reply:
I'm trying to build generator using CMake, and so far it's testing well.

Reply:
There seems to be a problem with the generated_cpp so far, can you take a look at it?
On the first compile it will run the GenerateCpp target, on the second it will use the automatically generated generated_cpp directory, but there are some symbol problems